### PR TITLE
Initial pull request

### PR DESCRIPTION
**Pull request by jinimcoroneo**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'sinatra/base'
+require './lib/player'
+require './lib/game'
+
+class RPS < Sinatra::Base
+  enable :sessions
```
Comment (maryrosecook) [inline app.rb]: Do sessions need to be enabled?

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @game = Game.create(Player.new(params[:player_name]))
+    redirect '/welcome'
+  end
+
+  get '/welcome' do
+    erb :welcome
+  end
+
+  post '/play' do
+    @game.player.choose(params[:choice]).to_sym
```
Comment (maryrosecook) [inline app.rb]: Could the player's move be set without breaking the encapsulation of `game`?

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  before do
+    @game = Game.instance
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  post '/names' do
```
Comment (maryrosecook) [inline app.rb]: Love these clearly named routes and correspondingly clearly named templates: names, welcome, play

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  def self.instance
+    @game
+  end
+
+  def initialize(player, opponent_choice = nil)
+    @player = player
+    @opponent_choice = opponent_choice
+  end
+
+  def choose
```
Comment (maryrosecook) [inline lib/game.rb]: Does the name `choose` reveal the intention of this method?

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
+require "./lib/player.rb"
+
+class Game
+
+  GAME_RULES = {
```
Comment (maryrosecook) [inline lib/game.rb]: Does the name `GAME_RULES` reveal the intention of this hash? What data is it storing? What name would help the most for someone unfamiliar with the code?

File: lib/game.rb (hunk 4 above)
Comment (maryrosecook) [inline lib/game.rb]: Is this behaviour in the right place? Does a game of RPS know how to generate a random move? Or could that behaviour be moved somewhere else?

File: lib/game.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  def initialize(player, opponent_choice = nil)
+    @player = player
+    @opponent_choice = opponent_choice
+  end
+
+  def choose
+    @opponent_choice = [:rock, :paper, :scissors].sample
+  end
+
+  def win?
```
Comment (maryrosecook) [inline lib/game.rb]: Love these domain-revealing helper methods!

File: lib/player.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+  attr_reader :name, :choice
+
+  def initialize(name, choice = nil)
+    @name = name
+    @choice = choice
+  end
+
+  def choose(choice)
+    @choice = choice.to_sym
```
Comment (maryrosecook) [inline lib/player.rb]: This is a nitpick, but should this method call `to_sym`, or should that be done in the controller, given that the controller understands how the view and model message each other?

File: spec/features/welcome_spec.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
+# As a marketeer
+# So that I can see my name in lights
+# I would like to register my name before playing an online game
+
+feature 'see my name on welcome page' do
```
Comment (maryrosecook) [inline spec/features/welcome_spec.rb]: Love all these clear tests!

File: views/play.erb [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      <form action="/welcome">
+      <input type="submit" class="button buttonblack" value="Play again?">
+      </form>
+    </div>
+
+  </body>
+</html>
+
+
+<style>
```
Comment (maryrosecook) [inline views/play.erb]: Is there a way to DRY up this styling code?

File: app.rb (hunk 2 above)
Comment (jinimcoroneo) [inline app.rb]: I'm a bit confused with regards to this.  I can only access the player via the game, is there another way to do this using dependency injection?

Comment (maryrosecook) [inline app.rb]: One option is to add a method to `Game` that will make the player choice. Another is to only create the game once the player's choice has been set.

**Pull request closed without merging**